geomstats/geometry/general_linear.py

```diff
@@ -200,17 +200,14 @@ class SquareMatrices(MatrixLieAlgebra):
 
     def __init__(self, n, equip=True):
         self.n = n
-        super().__init__(dim=n**2, representation_dim=n, equip=equip)
-        self._mat_space = Matrices(n, n, equip=False)
+        super().__init__(n=n, m=n, equip=equip)
 
-    @staticmethod
-    def default_metric():
-        """Metric to equip the space with if equip is True."""
-        return MatricesMetric
 
-    def _create_basis(self):
-        """Create the canonical basis of the space of matrices."""
-        return self._mat_space.basis
+class SquareMatricesLieAlgebraStruct(MatrixLieAlgebra):
```

Comment:
No need for "Struct" IMO. If we want to append something, let's use Structure.

Reply:
Even better, we could give this the name of the structure, i.e.
`SquareMatricesLieBracket`, so that:
- `VectorSpace` + `LieBracket` = `LieAlgebra`.

Reply:
I've appended structure to avoid having someone thinking that this is the Lie algebra of square matrices (the concept does not make sense, but it could lead to thinking something like that).

Nevertheless, your last comment fully answers it: very nice naming strategy!